### DISCUSSION TITLE
Trouble understanding block.size in VIMP

### DISCUSSION THREAD

**Discussion by KenleyData** [kogalur/randomForestSRC#217] [Q&A] [answered]
I am using RandomForestSRC for a survival analysis problem including calculating variable importance. My code works fine, but as I'm reading all of the documentation there is a piece that I don't understand, and I was hoping I could find clarification here (or referral to other documentation).

I'm confused about the concept of block.size. What I have learned from the documentation is that the prediction error is calculated for each block of trees, and then this prediction error is averaged over all blocks to give the number that is reported by VIMP. My confusion is that I thought prediction error was essentially a summed/averaged quantity itself, averaging over the value for each tree (in which case block.size wouldn't matter because averaging is averaging). Is the idea that all trees/cases in the block would be pooled together before assessing every possible pair in the Harrell's c-index? I can see how that wouldn't be a straightforward averaging over trees.

Thanks very much for any thoughts/clarification!

**Comment by ishwaran** [comment, marked as answer]
Yes, you are correct in your interpretation.  Setting `block.size=1` yields the traditional VIMP which is to average the VIMP for each tree over the forest.  So the number of VIMP values equals the number of trees and we average over these. Otherwise what happens is that blocks of trees, each block comprised of `block.size` trees, are used to construct a mini ensemble and the VIMP for each mini ensemble is calculated.  These mini ensemble VIMP values are averaged over the number of blocks.  In the limit when `block.size` is set to the number of trees, this amounts to calculating the VIMP for the ensemble.   So just one calculation is done.

Essentially it's a computational savings device especially useful for survival analysis when C-index is being used (which is very expensive for big data).

**Reply by KenleyData** [reply]
Thanks so much for your quick reply!

**Reply by ishwaran** [reply]
Also just to be clear.  It's a computational savings BUT also keep in mind that the values and properties of the resulting VIMP will also be different.  VIMP values depend strongly on` block.size`.
